apps/ejabberd/src/mod_vcard.erl

```diff
@@ -539,7 +541,7 @@ prepare_vcard_search_params(User, VHost, VCARD) ->
 
     US = {LUser, VHost},
 
-    if
+    case
```

Comment:
This function is executed always inside `try..catch` that returns `internal-server-error` to the client. I think it is good enough, as it's very difficult to trigger this error and client would have to send malformed data on purpose (or be *very* unlucky). Unfortunately XEP doesn't specify any specific error response for invalid VCard fields.

I'm going to remove this `case` and "let it crash". Do you agree?